### PR TITLE
[Vision] Add pdf text detection sample

### PR DESCRIPTION
**Pull request by alixhami**

### CODE REVIEW

**Comment by tbpg** [comment]
I cancelled Circle since it's not passing right now (focusing on migrating to Kokoro). The tests passed on Kokoro, so you're good on that front. :+1:

File: vision/detect_pdf.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 Google, Inc
```
Comment (frankyn) [inline vision/detect_pdf.rb]: @alixhami this is awesome, but not all vision samples have been migrated over to autogen library. This sample will most likely cause confusion with [existing samples](https://cloud.google.com/vision/docs/detecting-labels). 

@beccca is migration to Vision GAPIC prioritized by the ML APIs team for Q3?

**Comment by beccca** [comment]
@frankyn I'm giving this sample my 👍 I'm explicitly biased towards getting this functionality in users' hands to play with. But, wow, man oh man is this different! Yuck yuck yuck. The code works & is clean (for what it is) & provides access to a powerful, requested feature. But. Yeah. It is what it is.

I've added you to get an explicit LGTM if you're cool with this sample going in – to counterbalance my bias 😇

File: vision/detect_pdf.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+def detect_pdf_gcs gcs_source_uri:, gcs_destination_uri:, project_id:
+  # [START vision_async_detect_document_ocr]
+  # project_id = "Your Google Cloud project ID"
+  # gcs_source_uri = "Google Cloud Storage URI, eg. 'gs://my-bucket/example.pdf'"
+  # gcs_destination_uri = "Google Cloud Storage URI, eg. 'gs://my-bucket/prefix_'"
+
+  require "google/cloud/vision"
+  require "google/cloud/storage"
+
+  vision_client = Google::Cloud::Vision::V1.new
```
[the remark this answers is not in the document]
Comment (beccca) [reply vision/detect_pdf.rb]: @TheRoyalTnetennba please TAL – looks like Project doesn't delegate to rpc methods?

[the remark this answers is not in the document]
Comment (beccca) [reply vision/detect_pdf.rb]: ^---- @frankyn @TheRoyalTnetennba @alixhami This is my only major blocking issue

This doesn't make any sense to me... I thought Google::Cloud::Vision.new was a factory constructor which delegates to Google::Cloud::Vision::[DEFAULT].new and the default is already V1

[the remark this answers is not in the document]
Comment (beccca) [reply vision/detect_pdf.rb]: @TheRoyalTnetennba As Vision is still hand-written, would it be possible to quickly add a delegation from Project --> delegate just this async_batch_annotate_files method to the service?

(FYI there are no other API methods which Project does not support, #annotate supports everything _except_ this new method. A GAPIC transition needs to happen for this library but I'd like to advocate for this small change as a crutch in the meantime – I don't expect any other additions to this library EVER until it's GAPIC-ified)

**Comment by beccca** [comment]
Update: pending [google-cloud-ruby PR2198](https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2198) & gem update

**Comment by beccca** [comment]
Unblocked! Thanks to @TheRoyalTnetennba & @dazuma a new version of the library was published with this functionality (`google-cloud-vision 0.30.0`)

@alixhami you can update the gem and now safely make the change from `Google::Cloud::Vision::V1.new` to `Google::Cloud::Vision.new`

**Pull request merged**